### PR TITLE
Fix DB asset upgrade when custom assets owned by the user are also added by the upgrade

### LINKED ISSUE #2791: User ended up with a corrupt token entry in the DB after conflict during upgrade
## Problem Definition

A user contacted us via email and shared some screenshots which gave funny errors when trying to find/insert a token which should have been added via the v1 assets upgrade. The user said they had it already locally but opted to solve the confict by choosing remote.

I got their global.db and it seems that something is wrong.

```
sqlite> SELECT * from assets WHERE identifier="_ceth_0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b";
_ceth_0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b|C|Rally|RLY|||rally-2||0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b

sqlite> SELECT * from ethereum_tokens where address="0xf1f955016EcbCd7321c7266BccFB96c68ea5E49b";
sqlite> 
```
So the asset has an entry in the `assets` table and not in the `ethereum_tokens` table. 

## Task

Figure out how this could have happened and if it's something on our side fix.

### PR DESCRIPTION
**Pull request by LefterisJP**
Fix #2791

### CODE REVIEW

File: rotkehlchen/globaldb/updates.py [hunk 1]
```diff
@@ -261,6 +287,10 @@ def _apply_single_version_update(
     ) -> None:
         lines = text.splitlines()
         for action, full_insert in zip(*[iter(lines)] * 2):
+            # if '0x71F85B2E46976bD21302B64329868fd15eb0D127' in action:
+            #     __import__("pdb").set_trace()
+            #     a = 1
+
```
Comment (yabirgb) [inline rotkehlchen/globaldb/updates.py:293]: Is this a leftover?

File: rotkehlchen/globaldb/updates.py [hunk 2]
```diff
@@ -35,12 +37,36 @@ def executeall(cursor: sqlite3.Cursor, statements: str) -> None:
...
+    ) WHERE name="{ASSETS_VERSION_KEY}";
+    PRAGMA foreign_keys = ON;
+    DETACH DATABASE "other_db";
+    """)
+
+
 def _force_remote(cursor: sqlite3.Cursor, local_asset: Asset, full_insert: str) -> None:
     """Force the remote entry into the database by deleting old one and doing the full insert.
-
+σ
```
Comment (yabirgb) [inline rotkehlchen/globaldb/updates.py:66]: This sigma might not be intended

File: rotkehlchen/globaldb/updates.py [hunk 3]
```diff
@@ -35,12 +37,36 @@ def executeall(cursor: sqlite3.Cursor, statements: str) -> None:
...
     May raise an sqlite3 error if something fails.
     """
-    cursor.execute('PRAGMA foreign_keys=off;')
+    cursor.executescript('PRAGMA foreign_keys=off;')
```
Comment (yabirgb) [inline rotkehlchen/globaldb/updates.py:69]: Why is executescript used here? It makes a commit before executing the sql statement. As per discused this morning doesn't it need to be a commit after the pragma sentence?

File: rotkehlchen/globaldb/updates.py (hunk 2 above)
Comment (LefterisJP) [reply rotkehlchen/globaldb/updates.py:66]: hahahaahahahaahahah yes it's not.

File: rotkehlchen/globaldb/updates.py (hunk 1 above)
Comment (LefterisJP) [reply rotkehlchen/globaldb/updates.py:293]: ah yes good. thanks.

File: rotkehlchen/globaldb/updates.py (hunk 3 above)
Comment (LefterisJP) [reply rotkehlchen/globaldb/updates.py:69]: exactly because I need this to be commited.

Comment (LefterisJP) [reply rotkehlchen/globaldb/updates.py:69]: But need to confirm if executescript does a commit at the end or not. I thought it does. If it does not, then I will have to also add a `COMMIT` inside there too.

File: rotkehlchen/globaldb/updates.py (hunk 2 above)
Comment (LefterisJP) [reply rotkehlchen/globaldb/updates.py:66]: funny that no linter hit AT ALL for this. Also for the commented code. I was telling myself. Remove the commented out shit, don't forget.

File: rotkehlchen/globaldb/updates.py (hunk 3 above)
Comment (yabirgb) [reply rotkehlchen/globaldb/updates.py:69]: As per the docs it is a commit then sql https://docs.python.org/3/library/sqlite3.html#sqlite3.Cursor.executescript

Comment (LefterisJP) [reply rotkehlchen/globaldb/updates.py:69]: All right so I will add a commit in there too, just to keep it as one liner.

**Comment by LefterisJP** [comment]
All is green so I am going to merge this @yabirgb

**Comment by yabirgb** [comment]
Yes reviewed the last commit when you updated it. Seems good

**Pull request merged**